### PR TITLE
Sct converter based on v1.4dev

### PR DESCRIPTION
**Pull request by RPeschke**
LCIO Converter for the SCT data converter.

new generic function for convertion between EUDAQ planes and LCIO data container.

Changes to the lcio file writer to make it more compatable with the nreader 

added DLL Export for Detector description DLL

### CODE REVIEW

File: debug/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# ./debug
+*
+
+!.gitignore
```
Comment (simonspa) [inline debug/.gitignore]: This commit adds a folder "debug" that I'd rather not have in the repo. You can exclude the whole folder by changing the toplevel .gitignore.

File: main/lib/plugins/DataConverterPlugin.cc [hunk 2]
```diff
@@ -23,4 +28,16 @@ namespace eudaq {
...
+     // helper object to fill the TrakerDater object 
+    auto sparseFrame =eutelescope::EUTelTrackerDataInterfacerImpl<eutelescope::EUTelGenericSparsePixel>(&zsFrame);
+
+    for (size_t iPixel = 0; iPixel < plane.HitPixels(); ++iPixel) {
+      eutelescope::EUTelGenericSparsePixel thisHit1(plane.GetX(iPixel), plane.GetY(iPixel), plane.GetPixel(iPixel), 0);
+      sparseFrame.addSparsePixel(&thisHit1);
+    }
+  
+  }
+
```
Comment (simonspa) [inline main/lib/plugins/DataConverterPlugin.cc]: Does this in any way change the behaviour of existing LCIO conversion plugins?

**Comment by simonspa** [comment]
Concerning your changes to the nreader includes: there are some more files which did not get the DLLEXPORT added. Could that be done if this is necessary for Windows compilation?

**Comment by simonspa** [comment]
Build failed:
```
[ 54%] Building CXX object main/lib/CMakeFiles/EUDAQ.dir/plugins/SCTUpgradeConverterPlugin.cc.o
/home/travis/build/eudaq/eudaq/main/lib/plugins/SCTUpgradeConverterPlugin.cc:59:36: error: ‘LCCollectionVec’ was not declared in this scope
   bool Collection_createIfNotExist(LCCollectionVec** zsDataCollection, const lcio::LCEvent & lcioEvent, const char * name){
                                    ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/SCTUpgradeConverterPlugin.cc:59:54: error: ‘zsDataCollection’ was not declared in this scope
   bool Collection_createIfNotExist(LCCollectionVec** zsDataCollection, const lcio::LCEvent & lcioEvent, const char * name){
                                                      ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/SCTUpgradeConverterPlugin.cc:59:72: error: expected primary-expression before ‘const’
   bool Collection_createIfNotExist(LCCollectionVec** zsDataCollection, const lcio::LCEvent & lcioEvent, const char * name){
                                                                        ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/SCTUpgradeConverterPlugin.cc:59:105: error: expected primary-expression before ‘const’
   bool Collection_createIfNotExist(LCCollectionVec** zsDataCollection, const lcio::LCEvent & lcioEvent, const char * name){
                                                                                                         ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/SCTUpgradeConverterPlugin.cc:59:122: error: expression list treated as compound expression in initializer [-fpermissive]
   bool Collection_createIfNotExist(LCCollectionVec** zsDataCollection, const lcio::LCEvent & lcioEvent, const char * name){
                                                                                                                          ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/SCTUpgradeConverterPlugin.cc:59:123: error: expected ‘,’ or ‘;’ before ‘{’ token
   bool Collection_createIfNotExist(LCCollectionVec** zsDataCollection, const lcio::LCEvent & lcioEvent, const char * name){
                                                                                                                           ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/SCTUpgradeConverterPlugin.cc:385:1: error: expected ‘}’ at end of input
 } // namespace eudaq
 ^
make[2]: *** [main/lib/CMakeFiles/EUDAQ.dir/plugins/SCTUpgradeConverterPlugin.cc.o] Error 1
make[1]: *** [main/lib/CMakeFiles/EUDAQ.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by RPeschke** [comment]
Hi,

ConvertPlaneToLCIOGenericPixel:

the function is only used by my converter so far. therefore it is not changing anything. 

i made a fix can you compile it now ?

**Comment by simonspa** [comment]
Nope:
```
[ 45%] Building CXX object main/lib/CMakeFiles/EUDAQ.dir/plugins/DataConverterPlugin.cc.o
/home/travis/build/eudaq/eudaq/main/lib/plugins/DataConverterPlugin.cc:31:68: error: ‘lcio::TrackerDataImpl’ has not been declared
   void ConvertPlaneToLCIOGenericPixel(StandardPlane & plane, lcio::TrackerDataImpl& zsFrame)
                                                                    ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/DataConverterPlugin.cc: In function ‘void eudaq::ConvertPlaneToLCIOGenericPixel(eudaq::StandardPlane&, int&)’:
/home/travis/build/eudaq/eudaq/main/lib/plugins/DataConverterPlugin.cc:34:23: error: ‘eutelescope’ has not been declared
     auto sparseFrame =eutelescope::EUTelTrackerDataInterfacerImpl<eutelescope::EUTelGenericSparsePixel>(&zsFrame);
                       ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/DataConverterPlugin.cc:34:67: error: ‘eutelescope’ has not been declared
     auto sparseFrame =eutelescope::EUTelTrackerDataInterfacerImpl<eutelescope::EUTelGenericSparsePixel>(&zsFrame);
                                                                   ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/DataConverterPlugin.cc:37:7: error: ‘eutelescope’ has not been declared
       eutelescope::EUTelGenericSparsePixel thisHit1(plane.GetX(iPixel), plane.GetY(iPixel), plane.GetPixel(iPixel), 0);
       ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/DataConverterPlugin.cc:37:44: error: expected ‘;’ before ‘thisHit1’
       eutelescope::EUTelGenericSparsePixel thisHit1(plane.GetX(iPixel), plane.GetY(iPixel), plane.GetPixel(iPixel), 0);
                                            ^
/home/travis/build/eudaq/eudaq/main/lib/plugins/DataConverterPlugin.cc:38:35: error: ‘thisHit1’ was not declared in this scope
       sparseFrame.addSparsePixel(&thisHit1);
                                   ^
make[2]: *** [main/lib/CMakeFiles/EUDAQ.dir/plugins/DataConverterPlugin.cc.o] Error 1
make[1]: *** [main/lib/CMakeFiles/EUDAQ.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by RPeschke** [comment]
do you build with LCIO enabled?

**Comment by simonspa** [comment]
Nope, we don't. See here: https://github.com/eudaq/eudaq/blob/master/.travis.yml
Building with LCIO enabled would require to install EUTelescope/ILCSoft on the build machines every time.

**Comment by RPeschke** [comment]
ok this explains the error because there was a #ifdef missing. 

can you check it on any pc that has LCIO installed?

thanks
Richard

**Comment by simonspa** [comment]
Compiles fine with (enabling NReader) and without LCIO now.
Remove the debug folder, and then we can go for a merge I'd say.

**Comment by simonspa** [comment]
Ah, and the nreader DLLEXPORT issue.

**Comment by simonspa** [comment]
Missing:
EUTelPixelDetector.h
EUTelCMSPixelDetector.h
EUTelBaseDetector.h

**Pull request merged**